modules/compiler/targets/host/source/target.cpp

```diff
@@ -102,6 +102,29 @@ static llvm::TargetMachine *createTargetMachine(llvm::Triple TT,
       /*JIT=*/true);
 }
 
+void UpdateFeatureMapFromString(llvm::StringMap<bool> &FeatureMap,
```

Comment:
This looks like it duplicates some logic from LLVM's [`SubtargetFeatures`](https://llvm.org/doxygen/classllvm_1_1SubtargetFeatures.html) class that we could use.

Reply:
thanks, I've updated to use that.

Reply:
Ha, that's possibly a creative slight misuse of a very easy to accidentally misuse class? The constructor seems to assume that the features already have a `+`/`-` flag. When they don't, I'm not sure LLVM will keep the current behaviour of keeping subtly invalid feature strings, I'm not sure that's something we should rely on. Let me have a better look at this, maybe I'm worried over nothing, it does look as if the effect is exactly what we would want.

Reply:
I checked that it "worked" correctly without the +/- flag, and was able to test with hasFlag(), so i think it's okay.